components/booking/pages/BookingPage.tsx

```diff
@@ -42,9 +43,30 @@ const PhoneInput = dynamic(() => import("@components/ui/form/PhoneInput"));
 
 type BookingPageProps = BookPageProps | TeamBookingPageProps;
 
+type BookingFormValues = {
+  name: string;
+  email: string;
+  notes?: string;
+  locationType?: LocationType;
+  guests?: string[];
+  phone?: string;
+  customInputs?: {
+    [key: string]: string;
+  };
+};
+
 const BookingPage = (props: BookingPageProps) => {
   const { t, i18n } = useLocale();
   const router = useRouter();
+  const { contracts } = useContracts();
+
+  const { eventType } = props;
+  useEffect(() => {
+    if (eventType.metadata.smartContractAddress) {
+      if (!contracts[(eventType.metadata.smartContractAddress || null) as number]) router.replace("/");
```

Comment:
same as above: this needs to be `replace("/"+username);` => cal.com/rick

Reply:
and ideally don't use `as number`

this would be better:
`if (eventType.metadata.smartContractAddress && !contracts[eventType.metadata.smartContractAddress])`